docs/contributor/test-plan-unittesting.md

```diff
@@ -0,0 +1,128 @@
+# Unittesting Test Plan
+
+## **Purpose** 
+
+The unit testing phase focuses on maintainability of the code. This involves testing the logic of smallest grains of the code in isolation and it aims to prevent regression resulted by code changes. 
```

Comment:
```suggestion
The unit testing phase focuses on the maintainability of the code. This involves testing the logic of the smallest grains of the code in isolation and it aims to prevent regression resulting from code changes. 
```